hw/ip/i2c/data/i2c.hjson

```diff
@@ -15,52 +15,56 @@
   // INTERRUPT pins
   interrupt_list: [
     { name: "fmt_watermark"
-      desc: "raised when the FMT FIFO depth falls below the low watermark."
+      desc: "host mode interrupt: raised when the FMT FIFO depth falls below the low watermark."
     }
     { name: "rx_watermark"
-      desc: "raised if the RX FIFO is past the high watermark."
+      desc: "host mode interrupt: raised if the RX FIFO is past the high watermark."
     }
     { name: "fmt_overflow"
-      desc: "raised if the FMT FIFO has overflowed."
+      desc: "host mode interrupt: raised if the FMT FIFO has overflowed."
     }
     { name: "rx_overflow"
-      desc: "raised if the RX FIFO has overflowed."
+      desc: "host mode interrupt: raised if the RX FIFO has overflowed."
     }
     { name: "nak"
-      desc: "raised if there is no ACK in response to an address or data write"
+      desc: "host mode interrupt: raised if there is no ACK in response to an address or data write"
     }
     { name: "scl_interference"
-      desc: "raised if the SCL line drops early (not supported without clock synchronization)."
+      desc: "host mode interrupt: raised if the SCL line drops early (not supported without clock synchronization)."
     }
     { name: "sda_interference"
-      desc: "raised if the SDA line goes low when host is trying to assert high"
+      desc: "host mode interrupt: raised if the SDA line goes low when host is trying to assert high"
     }
     { name: "stretch_timeout"
-      desc: "raised if target stretches the clock beyond the allowed timeout period"
+      desc: "host mode interrupt: raised if target stretches the clock beyond the allowed timeout period"
     }
     { name: "sda_unstable"
-      desc: "raised if the target does not assert a constant value of SDA during transmission."
+      desc: "host mode interrupt: raised if the target does not assert a constant value of SDA during transmission."
     }
     { name: "trans_complete"
-      desc: "raised if the host terminates the transaction by issuing STOP or repeated START."
+      desc: '''
+        host and target mode interrupt.
+        In host mode, raised if the host terminates the transaction by issuing STOP or repeated START.
+        In target mode, raised if the external host issues a STOP or repeated START.
+      '''
```

Comment:
Ah let me have a look. Does it seem right to you though to have "whatever" this is triggered on both a re-start and stop? 
Because as you say a restart might not target the same device again.

On that note, I probably should also double check the FIFOs involved can be cleared by software.. in the event that we encounter this and there are still entries left over that were not consumed.

Reply:
Yeah, that seems fine.

In terms of personal style, I'm not whether I'd split them into separate events, but I suspect that doesn't matter. And besides, we put what actually happened in some FIFO, right? So there is at least some way to determine whether we have a combined format or the end of the transaction, and the application software can handle it accordingly.

Reply:
wait sorry, when you say separate events, what are you referring to? do you mean separate start and stop?
I've partially tweaked things this way to allow for easier removal of the repeated start acquisition fifo entry if that is something that makes sense to do. I wanted to loop back with the original authors to see if they had a specific case in mind.

Reply:
that's what i think too... because the driver should be well aware of that that "o i've seen a start...and now another start...without a stop".

Reply:
All four FIFOs can be cleared by writing 1s to i2c.FIFO_CTRL.

The hardware IP description describes recording a repeated start with a distinct symbol in the ACQ FIFO.

Reply:
yeah that's right, but i think @a-will and i aren't convinced the repeated start symbol is actually useful.
What do you think?

Reply:
alright! i'm going to merge for now, and circle back to change the interrupt name as Alex suggested (mostly because this changes a few other downstream things).
I will try to track down in parallel if there was another reason we wanted the repeated start recognition. 

Thanks for the reviews everyone.

Reply:
Could we use the name "i2c_read/write" or similar?